susemanager-utils/susemanager-sls/salt/switch_to_bundle/mgr_switch_to_venv_minion.sls

```diff
@@ -0,0 +1,25 @@
+{%- set avoid_venv_salt_minion = salt['pillar.get']('custom_info:mgr_avoid_venv_salt_minion') %}
+{%- set is_salt_ssh = salt['pillar.get']('contact_method') in ['ssh-push', 'ssh-push-tunnel'] %}
+{%- set is_saltboot = salt['file.file_exists']('/etc/ImageVersion') %}
+
+{%- if grains['saltversion'] == '3000' and not is_salt_ssh and                                                            
+       not avoid_venv_salt_minion == 'True' and not is_saltboot %}
+
+include:
+  - util.mgr_switch_to_venv_minion
+
+extend:
+  mgr_disable_salt_minion:
+    service.dead:
+      - require:
+        - sls: services.salt-minion
```

Comment:
I wonder why this is needed here, but not in the original state. Could this be a bug in the original state?

Reply:
With `state.show_highstate` this is what I get for `mgr_salt_minion_run` that is part of `services.salt-minion.sls`:
```
mgr_salt_minion_run:
        ----------
        __env__:
            base
        __sls__:
            services.salt-minion
        service:
            |_
              ----------
              name:
                  salt-minion
            |_
              ----------
              enable:
                  True
            |_
              ----------
              order:
                  last
            - running
  ```
So I assume that this is the service that's going to be enabled (in happens in the practice, the `salt-minion` service is run again after being disabled by the migration state and that's why I needed the `require`).

In `salt-minon.sls` we have:
```
{%- if salt['pkg.version']('venv-salt-minion') %}
{%- set salt_minion_name = 'venv-salt-minion' %}
{%- set susemanager_minion_config = '/etc/venv-salt-minion/minion.d/susemanager.conf' %}
{%- endif -%}
```
But `venv-salt-minion` is not installed at the time of the highstate rendering.

Reply:
Just to check that I understand it correctly, `salt-minion.sls` needs to be overridden by the migration because it would be applied later in the highstate and "revert" part of the migration?

If that's the case, I guess this would also happen if someone added the "original migration sls" to a highstate (which causes the inclusion of `salt-minion.sls`, something that does not happen when running the migration as a "single action"). Wouldn't it then make sense to have this `require` in the original migration sls to ensure the disabling happens after `salt-minion.sls` is applied?

Reply:
I believe that if someone added the original migration sls to the highstate that would also happen. Could be a good idea what you propose.

Reply:
I added the `require` in the original `util.mgr_switch_to_venv_minion`. I tried it and works.